Gradata/src/gradata/cloud/materializer.py

```
@@ -0,0 +1,375 @@
+- Tier 3 recognized but not applied.
+"""
+
+from __future__ import annotations
+
+import contextlib
+import json
+import logging
+import sqlite3
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Iterable
```

Ruff (0.15.10)

[warning] Import from `collections.abc` instead: `Iterable`

Import from `collections.abc`

(UP035)